src/client/include/com/amazonaws/kinesis/video/client/Include.h

```diff
@@ -1140,6 +1147,66 @@ struct __StreamMetrics {
 
     // Last measured transfer rate in bytes per second
     UINT64 currentTransferRate;
+
+    // V1 metrics following
+
+    // Total stream duration - stream uptime
+    UINT64 uptime;
+
+    // Total transferred bytes
+    UINT64 transferredBytes;
+
+    // Total number of streaming sessions including new/errored/active
+    UINT64 totalSessions;
+
+    // Total number of active streaming sessions - only the ones that actually have streamed some data
+    UINT64 totalActiveSessions;
+
+    // Average session duration
+    UINT64 avgSessionDuration;
+
+    // Total number of buffered ACKs
+    UINT64 bufferedAcks;
+
+    // Total number of received ACKs
+    UINT64 receivedAcks;
+
+    // Total number of persisted ACKs
+    UINT64 persistedAcks;
+
+    // Total number of error ACKs
+    UINT64 errorAcks;
+
+    // Total number of dropped frames
+    UINT64 droppedFrames;
+
+    // Total number of dropped fragments
+    // TODO: Near-realtime streaming mode is not used currently supported and this value will be 0
+    UINT64 droppedFragments;
+
+    // Total number of skipped frames
+    UINT64 skippedFrames;
+
+    // Total number of storage pressure events
+    UINT64 storagePressures;
+
+    // Total number of latency pressure events
+    UINT64 latencyPressures;
+
+    // Total number of Buffer pressure events
+    UINT64 bufferPressures;
+
+    // Total number of stream stale events
+    UINT64 staleEvents;
+
+    // Total number of put frame call errors
+    UINT64 putFrameErrors;
+
+    // Backend Control Plane API call latency which includes success and failure
+    UINT64 cplApiCallLatency;
+
+    // Backend Data Plane API call latency which includes success and failure
+    UINT64 dataApiCallLatency;

```

Comment:
Shouldn't this and control plane api latency only track the successful ones? There should be a separate failure/timeout metrics IMO.

Reply:
WOW - just noticed that I added a common API to calculate the latency but only invoking it in desctibe call. Let me update it. I also think that we should handle the erred cases too - for example the timeout.

RE: bufferPressures - this is tracking the number of times the buffer pressure events have been invoked. This is slightly different callback than latency pressure in sense that it looks at the overall buffer duration rather than MAX_LATENCY.